src/lustre/lustreArrayDependencies.ml

```diff
@@ -199,6 +199,10 @@ and process_expr ind_vars ctx ns proj indices expr =
       vars
     in
     R.ok graph
+  | ChooseOp (_, (_, i, _), e) -> 
+    let* graph = r e in
+    let graph = G.remove_vertex graph (i, [0])
+    in R.ok graph
```

Comment:
This code should be unreachable since choose operators are desugared before this point. Replace it with `assert false` and comment.

Reply:
addressed